src/sdft_implementations.jl

```diff
@@ -0,0 +1,47 @@
+export sdft, SDFT
+
+import .SlidingDFTs
+import .SlidingDFTs: sdft
+
+## Basic SDFT
+
+"""
+    SDFT(n)
+
+Basic method to compute a Sliding Discrete Fourier Transform of window length `n`,
+through the recursive formula:
+
+\$X_{i+1}[k] = e^{j2{\\pi}k/n} \\cdot (X_{i}[k] + x[i+n] - x[i])\$
+
+The transfer function for the `k`-th bin of this method is:
+
+\$H(z) = \\frac{1 - z^{-n}}{1 - e^{j2{\\pi}k/n} z^{-1}}\$
+
+## References
+
+Jacobsen, E. & Lyons, R. (2003). "The sliding DFT," *IEEE Signal Processing Magazine*, 20(2), 74-80.
+doi:[10.1109/MSP.2003.1184347](https://doi.org/10.1109/MSP.2003.1184347)
+"""
+struct SDFT{T,C}
+    n::T
+    factor::C
+end
+
+function SDFT(n)
+    factor = exp(2π*im/n)
+    SDFT(n, factor)
+end
+
+# Required functions
+
+SlidingDFTs.windowlength(method::SDFT) = method.n
+
+function SlidingDFTs.updatedft!(dft, x, method::SDFT{T,C}, state) where {T,C}
+    twiddle = one(C)
+    for k in eachindex(dft)
+        dft[k] = twiddle * (dft[k] + SlidingDFTs.nextdata(state) - SlidingDFTs.previousdata(state))
+        twiddle *= method.factor
```

Comment:
If you multiply here a twiddle (which could be ComplexF128) with a `factor` of Float64 the type is still 128bits but the effective precision is only 64 bits.
So better would be if `method.factor` had a higher precision (such as Float128) if the user wants it.

Reply:
Solved in [ad33874](https://github.com/bionanoimaging/FourierTools.jl/pull/49/commits/ad33874a9043a5fc20904c7225934a45363bf4c7)